packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -0,0 +1,1050 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import '../color_scheme.dart';
+import '../divider.dart';
+import '../icon_button.dart';
+import '../icons.dart';
+import '../ink_well.dart';
+import '../material_localizations.dart';
+import '../text_theme.dart';
+import '../theme.dart';
+
+import 'date_picker_common.dart';
+import 'date_utils.dart' as utils;
+
+const Duration _monthScrollDuration = Duration(milliseconds: 200);
+
+const double _dayPickerRowHeight = 42.0;
+const int _maxDayPickerRowCount = 6; // A 31 day month that starts on Saturday.
+// One extra row for the day-of-week header.
+const double _maxDayPickerHeight = _dayPickerRowHeight * (_maxDayPickerRowCount + 1);
+const double _monthPickerHorizontalPadding = 8.0;
+
+const int _yearPickerColumnCount = 3;
+const double _yearPickerPadding = 16.0;
+const double _yearPickerRowHeight = 52.0;
+const double _yearPickerRowSpacing = 8.0;
+
+/// Displays a grid of days for a given month and allows the user to select a date.
+///
+/// Days are arranged in a rectangular grid with one column for each day of the
+/// week. It provides controls to change the year and month that the grid is showing.
+///
+/// The calendar picker widget is rarely used directly. Instead, consider using
+/// [showDatePicker], which will create a dialog that uses this as well as provides
```

Comment:
which creates a Dialog that contains a [CalendarDatePicker] and provides an optional compact view where the user can enter a date as a line of text.